src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
+            except Exception as e:
+                print("Error while fetching Dataset Details for Model card.", e)
+                return DatasetDetails()
+        
+        def update_training_details(exp_config: dict, gs_config: dict) -> TrainingDetails:
+            """
+            Function to initialize TrainingDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (TrainingDetails): initialized training details object.
+            """
+            try:
+                hyperparams = {}
+                hyperparams["optimizer"] = exp_config["optimizer"]["config"]["params"]
+                return TrainingDetails( hyperparams = hyperparams, loss_func = exp_config["train_component"]["config"]["loss_fun_config"]["tag"], optimizer = exp_config["optimizer"]["config"]["optimizer_key"])
+            except Exception as e:
+                print("Error while fetching Training Details for Model card.", e)
+                return TrainingDetails()
+        
+        def update_evaluation_details(exp_config: dict) -> EvalDetails:
+            """
+            Function to initialize EvalDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (EvalDetails): initialized evaluation details object.
+            """
+            loss_funcs = []
+            metrics = []
+            try:
+                for loss_func_config in exp_config["eval_component"]["config"]["loss_funs_config"]:
+                    loss_funcs.append(loss_func_config["tag"])
+            
+                for metric_config in exp_config["eval_component"]["config"]["metrics_config"]:
+                    metrics.append({"name": metric_config["tag"], "params": metric_config["params"]})
+
+                return EvalDetails(loss_funcs = loss_funcs, metrics = metrics)
+            except Exception as e:
+                print("Error while fetching Eval Details for Model card.", e)
+                return EvalDetails()
+
+        try:
+            experiment_env = ExperimentEnvironment(system_env = ExperimentEnvironment.create_system_info())
+            model_details = update_model_details(grid_search_id = grid_search_id, gs_config = gs_config, model=model)
+            dataset_details = update_dataset_details(exp_config = exp_config)
+            training_details = update_training_details(exp_config = exp_config, gs_config = gs_config)
+            eval_details = update_evaluation_details(exp_config = exp_config)
+            model_card = ModelCard(
+                model_details = model_details,
+                dataset_details = dataset_details,
+                experiment_environment = experiment_env,
+                training_details = training_details,
+                eval_details =eval_details
+            )
```

Comment:
I like how you separated this into different private functions :+1:  It improves the readability a lot

Reply:
thank you. :)